openeo_processes_dask/process_implementations/cubes/resample.py

```diff
@@ -0,0 +1,79 @@
+import logging
+from typing import Union
+
+import odc.geo
+from odc.geo.geobox import resolution_from_affine
+from pyproj.crs import CRS, CRSError
+
+from openeo_processes_dask.process_implementations.data_model import RasterCube
+
+logger = logging.getLogger(__name__)
+
+resample_methods_list = [
+    "near",
+    "bilinear",
+    "cubic",
+    "cubicspline",
+    "lanczos",
+    "average",
+    "mode",
+    "max",
+    "min",
+    "med",
+    "q1",
+    "q3",
+]
+
+
+def resample_spatial(
+    data: RasterCube,
+    projection: Union[str, int] = None,
+    resolution: Union[str, int] = None,
```

Comment:
Why can this be a string?

Reply:
The projection can be a string if provided as WKT or PROJ string. If it's an int it should be an EPSG code.
The resolution maybe should be int and float instead of string?

Reply:
yeah my mistake, the resolution should be an int!